prow/plugins/dog/dog_test.go

```diff
@@ -0,0 +1,229 @@
+
+func (c fakePack) readDog() (string, error) {
+	return string(c), nil
+}
+
+func TestRealDog(t *testing.T) {
+	if !*human {
+		t.Skip("Real dogs disabled for automation. Manual users can add --human [--category=foo]")
+	}
+	if cat, err := dogURL.readDog(); err != nil {
+		t.Errorf("Could not read dog from %s: %v", dogURL, err)
+	} else {
+		fmt.Println(cat)
+	}
+}
+
+func TestFormat(t *testing.T) {
+	re := regexp.MustCompile(`\[!\[.+\]\(.+\)\]\(.+\)`)
+	basicURL := "http://example.com"
+	testcases := []struct {
+		name string
+		url  string
+		err  bool
+	}{
+		{
+			name: "basically works",
+			url:  basicURL,
+			err:  false,
+		},
+		{
+			name: "empty url",
+			url:  "",
+			err:  true,
+		},
+		{
+			name: "bad url",
+			url:  "http://this is not a url",
+			err:  true,
+		},
+	}
+	for _, tc := range testcases {
+		ret, err := dogResult{
+			URL: tc.url,
+		}.Format()
+		switch {
+		case tc.err:
+			if err == nil {
+				t.Errorf("%s: failed to raise an error", tc.name)
+			}
+		case err != nil:
+			t.Errorf("%s: unexpected error: %v", tc.name, err)
+		case !re.MatchString(ret):
+			t.Errorf("%s: bad return value: %s", tc.name, ret)
+		}
+	}
+}
+
+// Medium integration test (depends on ability to open a TCP port)
+func TestHttpResponse(t *testing.T) {
+	url := "http://localhost"
+	b, err := json.Marshal(&dogResult{
+		URL: url,
+	})
+	blob := string(b)
+	if err != nil {
+		t.Errorf("Failed to encode test data: %v", err)
+	}
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if strings.Contains(r.URL.Path, "bad-json") {
+			io.WriteString(w, `{"bad-blob": "not-a-url"}`)
+		}
+		io.WriteString(w, blob)
+	}))
+	defer ts.Close()
+	fc := &fakegithub.FakeClient{
+		IssueComments: make(map[int][]github.IssueComment),
+	}
+
+	comment := "/woof"
+
+	e := &github.GenericCommentEvent{
+		Action:     github.GenericCommentActionCreated,
+		Body:       comment,
+		Number:     5,
+		IssueState: "open",
+	}
+	if err := handle(fc, logrus.WithField("plugin", pluginName), e, realPack(ts.URL)); err != nil {
+		t.Errorf("didn't expect error: %v", err)
+		return
+	}
+	if len(fc.IssueComments[5]) != 1 {
+		t.Error("should have commented.")
+		return
+	}
+	if c := fc.IssueComments[5][0]; !strings.Contains(c.Body, url) {
+		t.Errorf("missing image url: %s from comment: %v", url, c)
+	}
+
+	if err := handle(fc, logrus.WithField("plugin", pluginName), e, realPack(ts.URL+"/bad-json")); err == nil {
+		t.Errorf("expected an error")
+		return
+	}
+}
+
+// Small, unit tests
+func TestDogs(t *testing.T) {
+	var testcases = []struct {
+		name          string
+		action        github.GenericCommentEventAction
+		body          string
+		state         string
+		pr            bool
+		shouldComment bool
+		shouldError   bool
+	}{
+		{
+			name:          "ignore edited comment",
+			state:         "open",
+			action:        github.GenericCommentActionEdited,
+			body:          "/woof",
+			shouldComment: false,
+			shouldError:   false,
```

Comment:
Seems like shouldError is always false and therefore unnecessary?

Reply:
done